indico/modules/categories/controllers/base.py

```diff
@@ -52,6 +64,31 @@ def _checkProtection(self):
                            .format(self.category.no_access_contact))
             raise Forbidden(' '.join(msg))
 
+    @staticmethod
+    def format_event_date(event):
+        day_month = 'dd MMM'
+        if event.start_dt.year != event.end_dt.year:
+            return '{} - {}'.format(format_date(event.start_dt), format_date(event.end_dt))
+        elif (event.start_dt.month != event.end_dt.month) or (event.start_dt.day != event.end_dt.day):
+            return '{} - {}'.format(format_date(event.start_dt, day_month), format_date(event.end_dt, day_month))
+        else:
+            return format_date(event.start_dt, day_month)
+
+    def group_by_month(self, events):
```

Comment:
Is the base display class the best place for these methods? could have a `class RHDisplayCategoryEventsBase(RHDisplayCategoryBase)` that contains them and then have the two RHs that display event lists inherit from that one